v4/export/writer_util.go

```diff
@@ -114,6 +122,52 @@ func (b *writerPipe) ShouldSwitchStatement() bool {
 		(b.statementSizeLimit != UnspecifiedSize && b.currentStatementSize >= b.statementSizeLimit)
 }
 
+// WriteSpeedLimiter used for control speed of dump data
+type WriteSpeedLimiter interface {
+	CheckSpeed(*tcontext.Context, uint64)
+}
+
+// For no limit.
+type noWriteSpeedLimit struct{}
+
+func (sl *noWriteSpeedLimit) CheckSpeed(_ *tcontext.Context, _ uint64) {}
+
+// For limit.
+type rateWriteSpeedLimit struct {
+	limiter *rate.Limiter
+	limit   uint64
+}
+
+// The number of bytes currently downloaded has exceeded the speed limit value and needs to be paused.
+// Use a sliding window to prevent newSize from exceeding WaitN n.
+func (sl *rateWriteSpeedLimit) CheckSpeed(pCtx *tcontext.Context, newSize uint64) {
+	temp := newSize
+	for {
+		if temp <= sl.limit {
+			if err := sl.limiter.WaitN(pCtx, int(temp)); err != nil {
+				pCtx.L().Error("fail to write speed limit.", zap.Error(err))
+			}
+			break
+		}
+		if err := sl.limiter.WaitN(pCtx, int(sl.limit)); err != nil {
+			pCtx.L().Error("fail to write speed limit.", zap.Error(err))
```

Comment:
Check where this is a context error here. If so, return directly.
Or we may get blocked in this function.

Reply:
resolved.